apps/dialogue/src/dialogue.speech.service.ts

```diff
@@ -45,19 +45,26 @@ export class DialogueSpeechService {
     private readonly monitor: MonitorService,
   ) {}
 
-  async textToSpeech(payload: DialogueTextToSpeechDto): Promise<Buffer> {
-    return await this.ttsProvider.generateTTS(payload);
-  }
-
   async translateMessage(
     payload: DialogueMessageDto,
-    targetLanguage?: string,
+    toLanguage?: string,
   ): Promise<string> {
-    if (!targetLanguage) {
-      targetLanguage = await this.session.getLanguage(payload, false);
+    if (!toLanguage) {
+      toLanguage = await this.session.getLanguage(payload, false);
     }
 
-    if (!targetLanguage) return payload.text;
+    if (!toLanguage) return payload.text;
+
+    const fromLanguage = payload.text;
```

Comment:
Isn't payload.text the message to be translated? Are you using it as a language ISO code?

Reply:
gotcha, eagle eyes! Thanks!!!